embassy-net/src/lib.rs

```diff
@@ -416,7 +416,7 @@ impl<D: Driver> Stack<D> {
     /// ```ignore
     /// let config = embassy_net::Config::dhcpv4(Default::default());
     ///// Init network stack
-    /// static RESOURCES: StaticCell<embassy_net::StackResources<2>> = StaticCell::new();
+    /// static RESOURCES: StaticCell<embassy_net::StackResources<3>> = StaticCell::new();
```

Comment:
Maybe add comment why 3?

Reply:
done

Reply:
What happens if you don't have enough resources allocated, from a user perspective?
Does the application just panic's or do the tasks fight for a free socket?

Reply:
it panics.

Note this PR doesn't change the behavior, it just fixes many examples that had socket count set to 2 from before we had DNS (so it was enough, one for DHCP and one for the user). Now that we have DNS they need 3.